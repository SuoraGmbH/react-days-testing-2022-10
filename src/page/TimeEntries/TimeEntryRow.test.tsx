import { render, screen } from "@testing-library/react";
import TimeEntryRow from "./TimeEntryRow";
import TimeEntry from "../../domain/TimeEntry";
const timeEntry: TimeEntry = {
  id: "dousoijdsjiodsa",
  startTime: "13:30",
  endTime: "15:00",
  projectId: "2",
  date: "2022-10-20",
  comment: "Anfangen React Testing zu lernen",
};

describe("<TimeEntryRow />", () => {
  test("smoke detector ðŸ”¥", () => {
    render(
      <table>
        <tbody>
          <TimeEntryRow timeEntry={timeEntry} />
        </tbody>
      </table>
    );
  });

  test("renders the comment", () => {
    // ARRANGE
    render(
      <table>
        <tbody>
          <TimeEntryRow timeEntry={timeEntry} />
        </tbody>
      </table>
    );

    // ASSERT
    expect(screen.queryByText(/anfangen react testing/i)).toBeInTheDOM();
  });

  test("renders according to full snapshot", () => {
    const { container } = render(
      <table>
        <tbody>
          <TimeEntryRow timeEntry={timeEntry} />
        </tbody>
      </table>
    );

    expect(container).toMatchSnapshot();
  });

  test("renders according to text content snapshot", () => {
    const { container } = render(
      <table>
        <tbody>
          <TimeEntryRow timeEntry={timeEntry} />
        </tbody>
      </table>
    );
    expect(container.textContent).toMatchInlineSnapshot(
      `"22022-10-20ANFANGEN REACT TESTING ZU LERNEN"`
    );
  });
});
