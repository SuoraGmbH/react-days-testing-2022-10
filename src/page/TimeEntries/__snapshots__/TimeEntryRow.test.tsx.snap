// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeEntryRow /> renders according to full snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="pl-4 pr-3 font-medium whitespace-nowrap py-4 text-sm  text-gray-900 sm:pl-6"
        >
          2
        </td>
        <td
          class="px-3 whitespace-nowrap py-4 text-sm  text-gray-900 sm:pl-6"
        >
          2022-10-20
        </td>
        <td
          class="px-3 whitespace-nowrap py-4 text-sm  text-gray-900 sm:pl-6"
        >
          ANFANGEN REACT TESTING ZU LERNEN
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
